### PR TITLE
fix: are_all_blocked() returns false if an empty target list is provided

### PR DESCRIPTION
**Pull request by matthme**
Fixes a bug in the MemBlocks implementation.

## Summary by CodeRabbit

- Bug Fixes
  - Corrected block evaluation for empty target lists: an empty input now returns “not blocked” instead of being treated as fully blocked, preventing unintended blocking in edge cases.

- Tests
  - Added an asynchronous test to verify the corrected behavior for empty target lists, ensuring this edge case remains covered.

### CODE REVIEW

**Review by cdunster — COMMENTED** [review]
Good catch :+1:

Just want to mention the commit message doesn't really work with the changelog because the first character of the scenence is capitalised so it becomes:

> Are_all_block() returns false [...]

It might be good to start with something other than a function name or to encase it in monospace ticks

> fix: `are_all_blocked()` [...]

> fix: MemBlock::are_all_blocked() [...]

> fix: when an empty target list is passed to `are_all_blocked` then it incorrectly returned false

File: crates/core/src/factories/mem_blocks.rs [hunk 1]
```diff
@@ -78,6 +78,9 @@ impl Blocks for MemBlocks {
         &self,
         targets: Vec<BlockTarget>,
     ) -> BoxFut<'static, K2Result<bool>> {
+        if targets.is_empty() {
+            return Box::pin(async move { Ok(false) });
```
[the remark this answers is not in the document]
Comment (cdunster) [reply crates/core/src/factories/mem_blocks.rs:82]: This was actually from the Rabbit: https://github.com/holochain/kitsune2/pull/321#pullrequestreview-3165129513

but I think it's a valid point that the `move` isn't necessary

**Review by ThetaSinner — APPROVED** [review]
